### PR TITLE
BUG: Fix+test timezone-preservation in DTA.repeat

### PR DESCRIPTION
**Pull request by jbrockmendel**
Also added a whatsnew for #24265.

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1367,6 +1367,7 @@ Datetimelike
 - Bug in :attr:`Series.dt` where the cache would not update properly after an in-place operation (:issue:`24408`)
 - Bug in :class:`PeriodIndex` where comparisons against an array-like object with length 1 failed to raise ``ValueError`` (:issue:`23078`)
 - Bug in :meth:`DatetimeIndex.astype`, :meth:`PeriodIndex.astype` and :meth:`TimedeltaIndex.astype` ignoring the sign of the ``dtype`` for unsigned integer dtypes (:issue:`24405`).
+- Fixed bug in :meth:`Series.max` with ``datetime64[ns]``-dtype failing to return ``NaT`` when nulls are present and ``skipna=False`` is passed (:issue:`24265`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: this is only for tz aware right?

File: pandas/tests/arrays/test_datetimes.py [hunk 2]
```diff
@@ -90,3 +90,10 @@ def test_setitem_clears_freq(self):
                                         tz='US/Central'))
         a[0] = pd.Timestamp("2000", tz="US/Central")
         assert a.freq is None
+
```
Comment (jreback) [inline pandas/tests/arrays/test_datetimes.py]: is this near other repeat tests

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/test_datetimes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_datetimes.py]: There are no other repeat tests AFAICT; I think those are in the tests.extension part of 24024

File: doc/source/whatsnew/v0.24.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.24.0.rst]: the example in 24265 was tz-naive

File: pandas/tests/arrays/test_datetimes.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/arrays/test_datetimes.py]:
```
(pandas) bash-3.2$ grep -r test_repeat pandas/tests
pandas/tests/series/test_analytics.py:    def test_repeat(self):
pandas/tests/extension/base/methods.py:    def test_repeat(self, data, repeats, as_series, use_numpy):
pandas/tests/extension/base/methods.py:    def test_repeat_raises(self, data, repeats, kwargs, error, msg, use_numpy):
Binary file pandas/tests/extension/base/__pycache__/methods.cpython-36.pyc matches
pandas/tests/io/test_stata.py:    def test_repeated_column_labels(self):
pandas/tests/test_multilevel.py:    def test_repeat(self):
pandas/tests/test_strings.py:    def test_repeat(self):
pandas/tests/indexes/multi/test_reshape.py:def test_repeat():
pandas/tests/indexes/period/test_period.py:    def test_repeat_freqstr(self, index, use_numpy):
pandas/tests/indexes/common.py:    def test_repeat(self):
pandas/tests/indexes/datetimes/test_ops.py:    def test_repeat_range(self, tz_naive_fixture):
pandas/tests/indexes/datetimes/test_ops.py:    def test_repeat(self, tz_naive_fixture):
pandas/tests/indexes/timedeltas/test_ops.py:    def test_repeat(self):
pandas/tests/indexes/test_base.py:    def test_repeat(self):
```

Comment (jbrockmendel) [reply pandas/tests/arrays/test_datetimes.py]: right, i meant repeat tests for DTA/TDA/PA.  There are more coming up in the tests.extension part of 24024.

Comment (jreback) [reply pandas/tests/arrays/test_datetimes.py]: ok, i guess that 's fine. need a followup then to consolidate DTI/DTA test for common methods at some point then.

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
looks like this survived with needing rebase

**Comment by gfyoung** [comment]
> looks like this survived with needing rebase

Nice.  Thanks @jbrockmendel !

**Pull request merged**